src/SFA.DAS.PushNotifications.Functions/SFA.DAS.PushNotifications.Data/Repositories/ApplicationClientRepository.cs

```diff
@@ -0,0 +1,54 @@
+﻿using SFA.DAS.PushNotifications.Data.Entities;
+using static SFA.DAS.PushNotifications.Data.Entities.ApplicationClientStatusEnum;
+using static SFA.DAS.PushNotifications.Data.Entities.ApplicationEnum;
+
+namespace SFA.DAS.PushNotifications.Data.Repositories;
+
+public interface IApplicationClientRepository
+{
+    Task<int> AddWebPushNotificationSubscription(ApplicationClient applicationClient, CancellationToken cancellationToken);
+    Task RemoveWebPushNotificationSubscription(ApplicationClient applicationClient, CancellationToken cancellationToken);
+}
+
+public class ApplicationClientRepository : IApplicationClientRepository
+{
+    private readonly IPushNotificationsDataContext _context;
+
+    public ApplicationClientRepository(IPushNotificationsDataContext context)
+    {
+        _context = context;
+    }
+
+    public async Task<int> AddWebPushNotificationSubscription(ApplicationClient applicationClient, CancellationToken cancellationToken)
+    {
+        var appClient = _context.ApplicationClients.Where(x => x.Endpoint == applicationClient.Endpoint).FirstOrDefault();
```

Comment:
The Where filter clause can be moved to the filter in the FirstOrDefault() statement